src/server/game/Movement/MovementGenerator.h

```diff
@@ -44,6 +44,11 @@ class MovementGenerator
 
     virtual void unitSpeedChanged() { }
 
+    // timer in ms
+    virtual void Pause(uint32 /* timer = 0*/) {}
```

Comment:
Still does nothing. They are only virtual functions.

Reply:
im not really certain if it doesnt do nothing, tc has it set up the same way and with out the virtual function it will not compile.

Reply:
They use it to set pause timer for waypoint movement (when player interactswith NPC) ,but not sure about it.

Reply:
> 

i tested it walking vendors west of brill, mush room vendor, and walking guard and quest giver npcs on the same path with no issues, they do pause as intended so far from what i can tell and see now